x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/test_pipeline.tsx

```diff
@@ -151,9 +151,31 @@ export const TestPipeline: React.FC = () => {
                     </strong>
                   </p>
                 </EuiText>
-                <EuiCodeBlock fontSize="m" isCopyable language="json" paddingSize="m">
-                  {`[{"_index":"index","_id":"id","_source":{"${sourceField}":"bar"}}]`}
-                </EuiCodeBlock>
+                {fieldMappings && (
+                  <EuiCodeBlock fontSize="m" isCopyable language="json" paddingSize="m">
+                    {`[{"_index":"index","_id":"id","_source":{${fieldMappings
+                      .map(
+                        (fieldMapping) =>
+                          `"${fieldMapping.sourceField}": "${i18n.translate(
+                            'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.test.sampleValue',
+                            {
+                              defaultMessage: 'REPLACE ME',
+                            }
+                          )}"`
```

Comment:
Code duplication and redundant calls in the iteration. Please extract this block (translate sample value then store in a constant) into a function to e.g. line 52, then reference that instead.